internal/controller/spinapp_controller.go

```diff
@@ -248,7 +248,7 @@ func (r *SpinAppReconciler) ensureCASecret(ctx context.Context, caSecretName, na
 	}
 
 	err := r.Client.Get(ctx, types.NamespacedName{Name: caSecretName, Namespace: namespace}, secret)
-	if !apierrors.IsNotFound(err) { // secret is not not found
+	if !apierrors.IsNotFound(err) { // secret is not found
```

Comment:
I think the additional "not" originally here is on purpose. I think this comment is confusing though and we should get rid of it entirely.